nef-plugin/Utils/Result+Extensions.swift

```diff
@@ -0,0 +1,11 @@
+//  Copyright © 2020 The nef Authors.
+
+import Foundation
+
+extension Result {
+    
+    @discardableResult
+    public func fold<B>(_ fe: (Failure) -> B, _ fa: (Success) -> B) -> Result<B, Failure> {
```

Comment:
This should not be called `fold`, as such function should return `B`, not a result. There is not a usual function to map and handle errors at the same time. If you still think it is needed, I'd rename it to `mapOrRecover` or something like that.

Reply:
In fact, I want a fold, not sure why I set the output to Result instead to `B`.